pkg/kernel/networkservice/connectioncontextkernel/ipcontext/ipaddress/common.go

```diff
@@ -129,14 +152,44 @@ func create(ctx context.Context, conn *networkservice.Connection, isClient bool)
 				WithField("duration", time.Since(now)).
 				WithField("netlink", "AddrAdd").Debug("completed")
 		}
-		return waitForIPNets(ctx, ch, l, ipNets)
+		return waitForIPNets(ctx, ch, l, toAdd)
 	}
 	return nil
 }
 
+func getIPAddrDifferences(netlinkHandle *netlink.Handle, l netlink.Link, new []*net.IPNet) ([]*net.IPNet, []*net.IPNet, error) {
+	toAdd := []*net.IPNet{}
+	toRemove := []*net.IPNet{}
```

Comment:
Note: you can append into nil slices
```suggestion
	var toAdd []*net.IPNet{}
	var toRemove []*net.IPNet{}
```